spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java

```diff
@@ -295,4 +297,20 @@ public static List<Expression> partitionMapToExpression(
   public static String toColumnName(NamedReference ref) {
     return DOT.join(ref.fieldNames());
   }
+
+  /**
+   * fetch latest snapshot based on the branch tip. We are still using currentSnapshot() when commit
+   * is to main branch to validate behavior has not changed.
+   *
+   * @param table to fetch snapshot from.
+   * @param branch used to fetch snapshot from a branch ref.
+   * @return Snapshot
+   */
+  public static Snapshot latestSnapshot(Table table, String branch) {
```

Comment:
Is this the right place for this? Seems really independent of Spark. What about using `SnapshotUtil`?